### PR TITLE
do not merge -- checkpoint alert signal on 'reset center'

### PR DESCRIPTION
**Pull request by gcmcnutt**
We add another configured channel that will send a 0.5sec pulse out on that channel whenever a reset center occurs.  This is typically used to play an audio on the TX 'camera centered' -- the signal is inband and should work on BT and direct trainer channel.

### CODE REVIEW

File: firmware/src/src/defines.h [hunk 1]
```diff
@@ -17,6 +17,7 @@
 #define CALCULATE_PERIOD 6666   // (us) 150hz IMU calculations
 #define PWM_FREQUENCY 50        // (ms) PWM Period
 #define UIRESPONSIVE_TIME 10000 // (ms) 10Seconds without an ack data will stop;
+#define RESET_PULSE_PERIOD 500000 // (us) pulse duration for the reset signal to TX
```
Comment (gcmcnutt) [inline firmware/src/src/defines.h:20]: oops i think this can be removed

File: firmware/src/src/sense.cpp [hunk 2]
```diff
@@ -433,13 +433,33 @@ void calculate_Thread()
...
         int panch = trkset.panCh();
+        int alertch = trkset.alertCh();
         if(tltch > 0)
             channel_data[tltch - 1] = trpOutputEnabled == true ? tiltout_ui : trkset.Tlt_cnt();
         if(rllch > 0)
             channel_data[rllch - 1] = trpOutputEnabled == true ? rollout_ui : trkset.Rll_cnt();
         if(panch > 0)
             channel_data[panch - 1] = trpOutputEnabled == true ? panout_ui : trkset.Pan_cnt();
 
+        // TODO what is trpOutputEnabled?
```
Comment (dlktdr) [inline firmware/src/src/sense.cpp:444]: It disables the Tilt/Roll and Pan on long button press. Was a feature requested to make it more like the Fatshark Trinity where the output would only enable/disable on long button presses.

File: firmware/src/src/defines.h (hunk 1 above)
Comment (dlktdr) [inline firmware/src/src/defines.h:20]: Is this a lingering edit. I agree it should be in trackersettings like you have it.

File: firmware/src/src/sense.cpp [hunk 3]
```diff
@@ -433,13 +433,33 @@ void calculate_Thread()
...
+            }
+            if (sendingresetpulse) {
+                channel_data[alertch - 1] = TrackerSettings::MAX_PWM;
+                pulsetimer += (float)CALCULATE_PERIOD / 1000000.0;
+                if(pulsetimer > TrackerSettings::RECENTER_PULSE_DURATION) {
+                    sendingresetpulse = false;
+                }
+            }
+        }
+
```
Comment (dlktdr) [inline firmware/src/src/sense.cpp:462]: Haven't actually tested but this looks good to me! Although I think it should be moved above tilt / roll / pan setting line 433, I know the GUI/FW shouldn't ever let you pick two channels the same but I believe TRP should always superseded any possible channels set the same, should it occur.

**Pull request closed without merging**